lib/index.js

```diff
@@ -888,6 +894,312 @@ module.exports = () => {
         throw new Error('ec_seckey_tweak_mul', ret);
       }
 
+      /**
+       * @summary check if a point is valid on the curve
+       * @argumentument {Uint8Array} point - point to check
+       * @returns {boolean} true if valid, false otherwise
+       */
+      function isPoint(point) {
+        if (!point || !(point instanceof Uint8Array)) {
+          throw new TypeError('point must be a Uint8Array');
+        }
+
+        const pointPtr = charStar(point);
+
+        if (point.length === 32) {
+          const validXonlyPoint = Module.ccall(
+            'ec_is_valid_xonly_pubkey',
+            'number',
+            ['number'],
+            [pointPtr]
+          );
+
+          freeMalloc();
+          return validXonlyPoint === 1;
+        } else {
+          const validEcPubkey = Module.ccall(
+            'ec_is_valid_ec_pubkey',
+            'number',
+            ['number', 'number'],
+            [pointPtr, point.length]
+          );
+          freeMalloc();
+          return validEcPubkey === 1;
+        }
+      }
```

Comment:
let's create a `ec_is_point` method in main.c that takes care of calling the correct method in order to abstract this on this glue file.